spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -500,14 +501,12 @@ def save_df_to_csv(dataframe, fname_out):
     :return:
     """
     if os.path.isfile(fname_out):
-        # Concatenate existing CSV file to the new CSV file
-        dataframe_old = pd.read_csv(fname_out)
-        dataframe = pd.concat([dataframe_old, dataframe], axis=0, ignore_index=True)
-        # Merge rows that have the same 'filename', 'compression_level', and 'slice(I->S)'
-        dataframe = dataframe.groupby(
-            [dataframe.iloc[:, 0], dataframe.iloc[:, 1], dataframe.iloc[:, 2]]
-        ).max()  # Use max for a tiebreaker (this should never happen unless the user computes the same metric twice)
-    dataframe.to_csv(fname_out, na_rep='n/a', index=False)
```

Comment:
I'm happy that we no longer need to use `iloc` (instead explicitly using the column names), and I'm happy we no longer have to specify `index=False` -- we have a proper index now thanks to `.from_records(index=)` + `read_csv(index_col=)`.

Reply:
Yes, I was motivated to do this change when I started messing around with pandas DataFrames in the interactive interpreter to understand #4172 better. Just seeing the index column with 0, 1, 2, ... appearing, and how we had to work around it, made me want to dig a bit deeper.